### DISCUSSION TITLE
Open collective project and goals: Rewrite shader system

### DISCUSSION THREAD

**Discussion by Aggror** [stride3d/stride#1201] [Contributors]
A rather large topic that require a lot of work is the rewrite of the shader system. When worked out in more detail, it can be used in the plan for requesting an Epic grant.

1. What needs to be done?
More info from this older thread: https://github.com/stride3d/stride/issues/14 

Google doc draft with our request: https://docs.google.com/document/d/1n1QHoybFy8WHGZXwHR_Ee4ZEXqselEejkITIIgF3cx4/edit?usp=sharing

**Comment by HydeNaut** [comment]
Hi Aggror, and great effort. My 2 cents about Megascan in the Epic grant list: The integration of the Megascans is interesting from a new user point of view, so they could use good assets immediately for their games. And maybe Epic would appreciate an Epic-owned product gaining wider visibility. There is an example of Megascan integration in vvvv https://youtu.be/AkInLnC944w?t=2873 that could give you an example of what that integration could be like. This brings me to two points

1. The integration could be fairly easy, according to the video, since there is a relatively straight forward json api that can be used. It then needs some UI decisions.
2. I haven't seen anyone mention a strong desire for such a tool in Stride Editor. There is certainly a discussion to be had about integration of asset markets into editor for the convenience of Stride Artists, and I think there could be a good moment for integration, but maybe we need to put the resources towards a new UI system and plugin system that developers could build the tool from?

**Reply by Aggror** [reply]
Thanks for the comment.

About the megascan option: I haven't seen any requests for this (in the Discord channels at least) for it specifically. However I do think it would help gain Epic's attention as we integrate in to their eco system. With that foothold it might be easier to persuade them for the other subjects too.

As for the plugin system: personally that is in my top 3. Too many people will find the stride source too intimidating to even attempt building customisations. Plugins could greatly reduce development time for useful features. Added it to the list. @manio143 already made this extensive post about it. Working out a summarized plan for the grant would still need to be done: https://github.com/stride3d/stride/issues/1120
